SwiftLeeds/Views/My Conference/TalkCell.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  TalkCell.swift
+//  SwiftLeeds
+//
+//  Created by Matthew Gallagher on 25/06/2022.
+//
+
+import SwiftUI
+
+struct TalkCell: View {
+    private let time: String
+    private let details: String
+    private let isNext: Bool
+    private let speaker: String?
+    private let company: String?
+    private let gradientColors: [Color]?
+
+    @Environment(\.colorScheme) var colorScheme
+
+    init(time: String, details: String, isNext: Bool = false, speaker: String? = nil, company: String? = nil, gradientColors: [Color]? = nil) {
+        self.time = time
+        self.details = details
+        self.isNext = isNext
+        self.speaker = speaker
+        self.company = company
+        self.gradientColors = gradientColors
+    }
+
+    var body: some View {
+        VStack(spacing: 6) {
+            timeLabel(time)
+
+            VStack(alignment: .leading, spacing: 8) {
+                if let speaker {
+                    VStack(alignment: .leading, spacing: 4) {
+                        Text(speaker)
+                            .fontWeight(.medium)
+
+                        if let company {
+                            Text(company)
+                                .fontWeight(.medium)
+                                .opacity(0.5)
+                        }
+                    }
+                }
+
+                HStack {
+                    Text(details)
+                        .fontWeight(.light)
+                    Spacer()
+                }
+            }
+            .padding(12)
```

Comment:
Let's get these paddings pulled out to a DesignConstant file or similar so we can do things like .padding(Padding.medium).

Even better if there's extensions for .mediumPadding()

Reply:
There are lots of different padding amounts, so worth having a discussion on how we want to handle these throughout the app so the constants make sense

Reply:
Consistency is 🔑 mainly!

If they're all different then we'll just end up with a sloppy looking app. We can define the common paddings throughout and then treat them as semantic little/medium/large, rather than starting to add 12,8,7, etc throughout.

It would be good to start with pulling out what we have now and giving them names, then using those as the baseline

Reply:
These should be added to the design and then we can use them safely, but currently the design shows lots of different paddings, so I tried to make them uniform as much as I could.

Reply:
@pdamonkey this isn't resolved

Reply:
Gotcha, that wasn't stated so it just looked like there was a lingering discussion with no resolution.

I commented below with the addition to the Figma to explicitly state what these are, so they can be pulled out. We don't have to pull everything out, but the common values should be stored somewhere as if we decided to change design later we don't have to find and replace a bunch of numbers that could have completely different intentions.

Again to re-iterate, I did say that they're a concept, so there's gonna be some that are like 11 or 10 or something because they've just not been pixel-peeped, hence guidance being in place:

https://github.com/SwiftLeeds/swiftleeds-ios/pull/3#issuecomment-1167640552

I have no doubt some of these will change, especially as it comes to text as we may want to fiddle a little with tracking and splitting up into sub-stacks with varying levels of spacing, but the common values will be used throughout.

In this instance, I'd at least expect `cellPadding` ( probably a better name than that 😓 ) to be pulled out, and ideally `screenPadding`.

Reply:
I noticed the different values on the design, so that's why I used a set value that fit them, but will add local constants later today when I can get to the code.  Still not sure why it had to be in this PR and hold up future work, but if that's what I need for a tick, then I'll get on it and we can work out where to put them properly at a later date.